src/ui/router/ApplicationRouterSpec.js

```diff
@@ -1,139 +1,86 @@
-import { createOpenMct, resetApplicationState } from 'utils/testing';
+import { createOpenMct, waitForRouter, resetApplicationState } from 'utils/testing';
 
 let openmct;
 let element;
 let child;
 let appHolder;
 let resolveFunction;
 
-let initialHash = '';
-
-xdescribe('Application router utility functions', () => {
-    beforeAll(done => {
+describe('Application router utility functions', () => {
+    beforeEach(done => {
         appHolder = document.createElement('div');
         appHolder.style.width = '640px';
         appHolder.style.height = '480px';
 
         openmct = createOpenMct();
         openmct.install(openmct.plugins.MyItems());
-        openmct.install(openmct.plugins.LocalTimeSystem());
-        openmct.install(openmct.plugins.UTCTimeSystem());
 
         element = document.createElement('div');
         child = document.createElement('div');
         element.appendChild(child);
 
-        openmct.on('start', done);
+        openmct.on('start', waitForRouter(done));
+
         openmct.start(appHolder);
 
         document.body.append(appHolder);
     });
 
-    afterAll(() => {
-        openmct.router.setHash(initialHash);
+    afterEach(() => {
         appHolder.remove();
 
         return resetApplicationState(openmct);
     });
 
-    it('has initial hash when loaded', (done) => {
-        let success;
-        resolveFunction = () => {
-            openmct.router.setLocationFromUrl();
-            success = window.location.hash !== null;
-            if (success) {
-                initialHash = window.location.hash;
-                expect(success).toBe(true);
-
-                openmct.router.removeListener('change:hash', resolveFunction);
-                done();
-            }
-        };
-
-        openmct.router.on('change:hash', resolveFunction);
+    it('has initial hash when loaded', () => {
+        expect(window.location.hash).not.toBe(null);
     });
 
-    it('The setSearchParam function sets an individual search parameter in the window location hash', (done) => {
-        let success;
+    it('The setSearchParam function sets an individual search parameter in the window location hash', () => {
         openmct.router.setSearchParam('testParam', 'testValue');
-        resolveFunction = () => {
-            success = window.location.hash.includes('testParam=testValue');
-            if (success) {
-                expect(success).toBe(true);
-
-                openmct.router.removeListener('change:hash', resolveFunction);
-                done();
-            }
-        };
 
-        openmct.router.on('change:hash', resolveFunction);
+        const searchParams = openmct.router.getAllSearchParams();
+        expect(searchParams.get('testParam')).toBe('testValue');
```

Comment:
This is no longer testing what it says it's testing. It's checking router params, it is not checking the location hash at all.